src/components/generics/date-range-picker.tsx

```diff
@@ -0,0 +1,98 @@
+import { Box } from "@mui/material";
+import { DatePicker } from "@mui/x-date-pickers";
+import { DateTime } from "luxon";
+import { useEffect, useState } from "react";
+import { getTimeDifferenceInDays } from "../../utils/time-utils";
+
+/**
+ * Component properties
+ */
+interface DateRangePickerProps {
+  dateTimeNow: DateTime;
+  setDates: Function;
+  initialStartDate?: DateTime;
+  initialEndDate?: DateTime;
+}
+/**
+ * Date range picker component
+ *
+ * @param props DateRangePickerProps
+ */
+const DateRangePicker = (props: DateRangePickerProps) => {
+  const { dateTimeNow, setDates, initialStartDate, initialEndDate } = props;
+  const [startDate, setStartDate] = useState<DateTime | null | undefined>(
+    initialStartDate ? initialStartDate : dateTimeNow
+  );
+  const [endDate, setEndDate] = useState<DateTime | null | undefined>(
+    initialEndDate ? initialEndDate : dateTimeNow
+  );
+
+  /**
+   * Set startDate and endDate on initial dates changing
+   */
+  useEffect(() => {
+    if (initialStartDate) {
+      setStartDate(initialStartDate);
+    } else {
+      setStartDate(dateTimeNow);
+    }
+    if (initialEndDate) {
+      setEndDate(initialEndDate);
+    } else {
+      setEndDate(dateTimeNow);
+    }
+  }, [initialStartDate, initialEndDate]);
```

Comment:
Is this useEffect necessary? I tested creating a vacation with this commented and the dates seemed to update/render fine.

Reply:
This useEffect is needed for: 
set the date values on edit (from selected vacationRequest),
reset the date values on cancel,
reset the date values on create.

Reply:
However, this logic might not be needed here or at all if I lifted the state up, don't know.

Reply:
I'm all for cleaner code, so if the functionality does not change and lines of code are reduced, then go for it! :D